lib/shoulda/matchers/active_model/validator.rb

```diff
@@ -49,7 +49,7 @@ def has_messages?
         end
 
         def messages_description
-          if has_messages?
+          if has_messages? && !record.valid?
```

Comment:
This still seems odd to me, I'm investigating why we have to do this. Seems odd that the record has error messages but then it's valid at the same time.